crates/codegen/runtime/generator/src/parser/codegen/keyword_scanner_definition.rs

```diff
@@ -1,22 +1,23 @@
+use std::rc::Rc;
+
 use codegen_language_definition::model;
 use proc_macro2::TokenStream;
 use quote::{format_ident, quote};
 
-use crate::parser::codegen::scanner_definition::ScannerDefinitionNodeCodegen as _;
+use crate::parser::codegen::scanner_definition::ScannerExt as _;
 use crate::parser::codegen::versioned::VersionedQuote;
-use crate::parser::grammar::{KeywordScannerDefinitionRef, ScannerDefinitionNode};
 
 pub trait KeywordScannerDefinitionCodegen {
```

Comment:
I think this should be `KeywordScannerDefinitionCodegenerator`. The short form saves a few chars but has the wrong grammatical form IMO.

Reply:
Unfortunately, for better or worse, trait names in Rust weren't following strict naming scheme and shortened names were used instead early on, also due to OCaml influence, from my understanding. In `std`, we now have names like `Clone, Copy, Eq, Ord, ToString` etc. and it was preferred over forms of `Visitable`, `Comparable` etc.

I realize that this is dangerously close to Java's names using 5+ words (it's de facto `trait Codegen` but specialized for these 4 different types) but since we're talking about the form and naming, I'd argue that `Codegenerator` is less correct form to be used as a trait name in Rust, in this context, regardless if it saves or costs more characters.